pkg/limits/ingest_limits.go

```diff
@@ -387,6 +387,42 @@ func (s *IngestLimits) GetAssignedPartitions(_ context.Context, _ *logproto.GetA
 	return &resp, nil
 }
 
+// ExceedsLimits implements the logproto.IngestLimitsServer interface.
+// It returns the number of active streams for a tenant and the status of requested streams.
+func (s *IngestLimits) ExceedsLimits(ctx context.Context, req *logproto.ExceedsLimitsRequest) (*logproto.ExceedsLimitsResponse, error) {
+	var (
+		lastSeenAt = time.Now()
+		// Use the provided lastSeenAt timestamp as the last seen time
+		recordTime = lastSeenAt.UnixNano()
+		// Get the bucket for this timestamp using the configured interval duration
+		bucketStart = lastSeenAt.Truncate(s.cfg.BucketDuration).UnixNano()
+		// Calculate the rate window cutoff for cleaning up old buckets
+		rateWindowCutoff = lastSeenAt.Add(-s.cfg.RateWindow).UnixNano()
+
+		newStreams = make(map[int32][]Stream)
+
+		limit = uint64(1000)
+	)
+
+	for _, stream := range req.Streams {
+		partitionID := int32(stream.StreamHash % uint64(s.cfg.NumPartitions))
+
+		if !s.partitionManager.Has(partitionID) {
+			continue
```

Comment:
We should reject this case, at least that's how I've described the expected behavior it in the issue. For example, suppose we send a stream to the wrong instance because we are working on a outdated `GetAssignedPartitions` response, the stream will be accepted. But we don't know that it should have been accepted as we sent it to the wrong pod that doesn't own this partition.